client/src/hooks/useUpdateServer.js

```diff
@@ -0,0 +1,29 @@
+import { useUpdateEffect } from 'react-use';
+
+export default function useUpdateServer(updateData, updateURL, setErrorMsgs, deps) {
```

Comment:
Consider moving the base URL logic into this function and only providing an endpoint as an argument.

Reply:
I'd rather leave this hook a bit more generic, any harm in leaving as is?

Reply:
No harm, but I still recommend it at a later date!